### PR TITLE
[AIRFLOW-2799] Filtering UI objects by datetime is broken

### PR DESCRIPTION
**Pull request by kevcampb**
This change treats the datetimes supplied by the UI as being in UTC. This only
works correctly where the system is using UTC throughout, but is an improvement
over the current behavior (crash).

Make sure you have checked _all_ steps below.

### Jira

- [X] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-2799
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

Yes there is a description

### Tests

No tests as this involves UI code. 

### Commits

- [X] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [X] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

Not required

### Code Quality

- [No] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

utils.py never passed flake8 in the first place

### CODE REVIEW

File: airflow/www/utils.py [hunk 1]
```diff
@@ -428,7 +437,26 @@ def __call__(self, field, **kwargs):
...
 class UtcFilterConverter(FilterConverter):
+    utcdatetime_filters = (UTCDateTimeEqualFilter, UTCDateTimeNotEqualFilter,
+                           UTCDateTimeGreaterFilter, UTCDateTimeSmallerFilter,
+                           UTCDateTimeBetweenFilter, UTCDateTimeNotBetweenFilter,
+                           FilterEmpty)
+
     @filters.convert('utcdatetime')
     def conv_utcdatetime(self, column, name, **kwargs):
-        return self.conv_datetime(column, name, **kwargs)
+        return [f(column, name, **kwargs) for f in self.utcdatetime_filters]
```
Comment (ashb) [inline airflow/www/utils.py:462]: These should all probably live in airflow/utils/sqlalchemy.py

**Comment by bolkedebruin** [comment]
Hi @kevcampb. Thanks for your contribution. Unfortunately, this is not a fix. It assumes that the system (and client) is running in UTC which it might not be. I don't think undefined behavior in case of non UTC is better than a crash. Obviously, a crash is not what we want, but it should be fixed properly. Also as it crashed now, please make sure to add a test to ensure that doesn't happen any more in the future.

I also miss a proper root cause analysis and therefore an approach to the right fix. Finally, it will need to pass the flake8 tests which are differential tests. The existing situation we would like to improve upon. Please also provide a proper commit message according to the guidelines.

**Comment by kevcampb** [comment]
Hi Bolke,

Afraid I'm not going to do those. I'll leave this patch for anyone who wants a semi-usable system.

Regards,
Kevin

File: airflow/www/utils.py (hunk 1 above)
Comment (ashb) [reply airflow/www/utils.py:462]: I was wrong on this comment btw.

**Comment by ashb** [comment]
Closing in favour of #4061

**Pull request closed without merging**